wallet-new/integration/AccountSpecs.hs

```diff
@@ -89,7 +101,7 @@ accountSpecs wRef wc =
                     , pmtSpendingPassword = Nothing
                     }
             amounts <- pick $ shuffle [1..5]
-            let addrAndAmount = zip (map (\(addr : _) -> addr) $ map accAddresses accs) amounts
+            let addrAndAmount = zip (map (\(addr :| _) -> addr) $ map fromList $ map accAddresses accs) amounts
```

Comment:
I'm confused.

`fromList` is going to call `error` if the list isn't empty. You need `nonEmpty :: [a] -> Maybe (NonEmpty a)` to do this safely. Currently, we throw `error` if any list in the list is empty. We can get this behavior back safely with `traverse`.

```haskell
traverse :: (a -> Maybe b) -> [a] -> Maybe [b]
nonEmpty :: [a] -> Maybe (NonEmpty a)
-- substitue `a` for `[a]` and `b` for `NonEmpty a`
traverse nonEmpty :: [a] -> Maybe [NonEmpty b]
```

So we could write:

```haskell
case traverse nonEmpty (map accAddresses accs) of
    Nothing -> -- How to handle this case? It means that one account did not have at least one address.
    Just addresses -> zip addresses amounts
```